### PR TITLE
Use PHPs production defaults for php.ini. Ensure Lagoon overrides are read afterwards.

### LINKED ISSUE #720: php-fpm php.ini still sets serialize_precision to 17 instead of the new -1 default
Even when using php > 7.1, the image's php.ini still has `serialize_precision = 17`

This was the default till 7.1 and should now be changed to `-1`. See https://www.php.net/manual/en/ini.core.php

### PR DESCRIPTION
**Pull request by seanhamlin**
Several key changes from the existing PHP configuration

`disable_functions` = (now empty, pcntl was disabled previously)
`serialize_precision` = -1 (was 17)
`realpath_cache_size` = 4096k (increased from 256k)
`max_input_vars` = 2000 (increased from 1000)

## Other key file changes

We now default to `php.ini-production` for `php.ini`. The hardcoded `php.ini` has been removed from the repo. This file was clearly out of date, and contained configuration that was no longer relevant, and also missed newer configuration.

 `00-lagoon-php.ini.tpl` is placed `$PHP_INI_DIR/conf.d/00-lagoon-php.ini`, this is unchanged behaviour, and will be read after the production `php.ini` is read.

## New PHP variables

```
${PHP_MAX_INPUT_TIME:-900}
${PHP_POST_MAX_SIZE:-2048M}
${PHP_UPLOAD_MAX_FILESIZE:-2048M}
${PHP_SESSION_COOKIE_SECURE:-0}
${PHP_SESSION_COOKIE_HTTPONLY:-0}
${PHP_SESSION_COOKIE_SAMESITE}
```

There will need to be a Lagoon PR created after this PR is merged to document the new environment variables. I made these, as these seems reasonable that people would want to change them, and the defaults we use are higher than PHP's defaults.

Also we should add documentation around how to create a new configuration file in `$PHP_INI_DIR/conf.d/` and how overrides work.

Closes #720, #635

### CODE REVIEW

File: images/php-fpm/entrypoints/70-php-config.sh [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+# configuration, we need to ensure it appears alphabetically after `php.ini`.
+# The filename `zz-php-lagoon.ini` was chosen to make this obvious.
+#
+# The default `php.ini` is seeded from the default production values PHP
+# recommends.
+#
+# @see https://github.com/php/php-src/blob/PHP-8.2/php.ini-production
+
+mv "$PHP_INI_DIR/conf.d/00-lagoon-php.ini.tpl" "$PHP_INI_DIR/conf.d/zz-php-lagoon.ini"
+ep "$PHP_INI_DIR/conf.d/zz-php-lagoon.ini"
```
Comment (tobybellwood) [inline images/php-fpm/entrypoints/70-php-config.sh:13]: this changes the current behaviour, and would mean that anyone who previously overrode our overrides in the old 00 now get their overrides overridden by us in zz.

File: images/php-fpm/8.1.Dockerfile [hunk 2]
```diff
@@ -38,8 +38,8 @@ ENV TMPDIR=/tmp \
 COPY check_fcgi /usr/sbin/
 COPY entrypoints /lagoon/entrypoints/
 
-COPY php.ini /usr/local/etc/php/
-COPY 00-lagoon-php.ini.tpl /usr/local/etc/php/conf.d/
+RUN cp "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/conf.d/php.ini"
+COPY 00-lagoon-php.ini.tpl "$PHP_INI_DIR/conf.d/"
```
Comment (tobybellwood) [inline images/php-fpm/8.1.Dockerfile:42]: 8.1 is copying these assets into the `$PHP_INI_DIR/conf.d` folder - the others into `$PHP_INI_DIR/`

File: images/php-fpm/entrypoints/70-php-config.sh (hunk 1 above)
Comment (seanhamlin) [reply images/php-fpm/entrypoints/70-php-config.sh]: Changed this back.

File: images/php-fpm/8.1.Dockerfile (hunk 2 above)
Comment (seanhamlin) [reply images/php-fpm/8.1.Dockerfile]: Nice catch!

File: images/php-fpm/8.0.Dockerfile [hunk 3]
```diff
@@ -38,8 +38,8 @@ ENV TMPDIR=/tmp \
 COPY check_fcgi /usr/sbin/
 COPY entrypoints /lagoon/entrypoints/
 
-COPY php.ini /usr/local/etc/php/
-COPY 00-lagoon-php.ini.tpl /usr/local/etc/php/conf.d/
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+COPY 00-lagoon-php.ini.tpl "$PHP_INI_DIR/"
```
Comment (tobybellwood) [inline images/php-fpm/8.0.Dockerfile:42]:
```suggestion
COPY 00-lagoon-php.ini.tpl "$PHP_INI_DIR/conf.d/"
```

File: images/php-fpm/8.2.Dockerfile [hunk 4]
```diff
@@ -38,8 +38,8 @@ ENV TMPDIR=/tmp \
 COPY check_fcgi /usr/sbin/
 COPY entrypoints /lagoon/entrypoints/
 
-COPY php.ini /usr/local/etc/php/
-COPY 00-lagoon-php.ini.tpl /usr/local/etc/php/conf.d/
+RUN mv "$PHP_INI_DIR/php.ini-production" "$PHP_INI_DIR/php.ini"
+COPY 00-lagoon-php.ini.tpl "$PHP_INI_DIR/"
```
Comment (tobybellwood) [inline images/php-fpm/8.2.Dockerfile:42]:
```suggestion
COPY 00-lagoon-php.ini.tpl "$PHP_INI_DIR/conf.d/"
```

File: helpers/docker-compose.yml [hunk 5]
```diff
@@ -117,6 +117,30 @@ services:
...
+  php-8-2-dev:
+    image: uselagoon/php-8.2-cli:latest
+    environment:
+      - LAGOON_ENVIRONMENT_TYPE=development
+    << : *default-user # uses the defined user from top
+
+  php-8-2-prod:
+    image: uselagoon/php-8.2-cli:latest
+    environment:
+      - LAGOON_ENVIRONMENT_TYPE=production
```
Comment (tobybellwood) [inline helpers/docker-compose.yml:141]: I'd be tempted here to test overriding a couple of variables via env vars and checking the overrides have replaced the defaults.

**Review by tobybellwood — APPROVED** [review]
This is nice - good to have the tests in there too!

**Pull request merged**